File: LibCommon/FileSystem/FileInfo.cpp

#include "FileInfo.h"
#include <System/Environment/Unicode.h>

namespace Common::FileSystem
{
    FileInfo::FileInfo(Efi::FileInfo* info)
		: Size(info->Size), FileSize(info->FileSize), PhysicalSize(info->PhysicalSize),
		CreateTime(info->CreateTime), LastAccessTime(info->LastAccessTime),
		ModificationTime(info->ModificationTime), Attribute(info->Attribute), FileName(info->FileName)
	{
	}

    FileInfo FileInfo::Create(Efi::FileInfo* info)
	{
		if (info == nullptr)
			return Empty_FileInfo;

		return FileInfo(info);
	}
}

File: UEFISharp/Protocols/IO/Media/FileInfo.cs
using System.Runtime.InteropServices;

namespace EFI.Protocols.IO.Media;

[StructLayout(LayoutKind.Sequential)]
public struct FileInfo
{
    public ulong Size;
    public ulong FileSize;
    public ulong PhysicalSize;
    public Time CreateTime;
    public Time LastAccessTime;
    public Time ModificationTime;
    public FileAttributes Attribute;
    public nint FileName; // CHAR16*
}
